chore: Use only pre-commit hooks (#2057)
XRPLF/clio: CONTRIBUTING.md

3. Write and test your code.
4. Ensure that your code compiles with the provided build engine and update the provided build engine as part of your PR where needed and where appropriate.
5. Where applicable, write test cases for your code and include those in the relevant subfolder under `tests`.

> 6. Ensure your code passes [automated checks](#pre-commit-hooks)

Author: @maria-robobug could you please take a look at this file?

7. Squash your commits (i.e. rebase) into as few commits as is reasonable to describe your changes at a high level (typically a single commit for a small change). See below for more details.
8. Open a PR to the main repository onto the _develop_ branch, and follow the provided template.

> **Note:** Please read the [Style guide](#style-guide).

### `pre-commit` hooks

To ensure code quality and style, we use [`pre-commit`](https://pre-commit.com/).

> Please run the following command in order to use `pre-commit` hooks that are helpful for `clio` development:

Reviewer: ```suggestion
Run the following command to enable `pre-commit` hooks that help with Clio development:
```

```bash
pip3 install pre-commit
pre-commit install
```

`pre-commit` takes care of running each tool in [`.pre-commit-config.yaml`](https://github.com/XRPLF/clio/blob/develop/.pre-commit-config.yaml) in a separate environment.

> `pre-commit` will also attempt to automatically use `doxygen` to verify that everything public in the codebase is covered by doc comments.

Reviewer: ```suggestion
`pre-commit` also attempts to automatically use Doxygen to verify that everything public in the codebase has doc comments.
```

> If `doxygen` is not installed, the hook will raise a warning suggesting to install `doxygen` for future commits.

Reviewer: ```suggestion
If Doxygen is not installed, the hook issues a warning and recommends installing Doxygen for future commits.
```

### Formatting

> Code must conform to `clang-format`, unless the result would be unreasonably difficult to read or maintain.

Reviewer: ```suggestion
Code must conform to `clang-format`, unless the result is unreasonably difficult to read or maintain.
```

> In most cases the `pre-commit` hook will take care of formatting and will fix any issues automatically.

Reviewer: ```suggestion
In most cases the `pre-commit` hook takes care of formatting and fixes any issues automatically.
```

> To manually format your code, use `pre-commit run clang-format --files <your changed files>` for C++ files and `pre-commit run clang-format --files <your changed files>` for CMake files.

Reviewer: ```suggestion
To manually format your code, run `pre-commit run clang-format --files <your changed files>` for C++ files, and `pre-commit run cmake-format --files <your changed files>` for CMake files.
```

Author: This one I had to commit manually, because I pushed a new commit changing this line while you were looking 🙂

Reviewer: No worries! Docs LGTM 👍🏾
